### PR TITLE
feat: `avm/res/compute/disk`

### PR DESCRIPTION
**Pull request by segraef**
[avm.res.compute.disk](https://github.com/segraef/bicep-registry-modules/actions/workflows/avm.res.compute.disk.yml)

### CODE REVIEW

**Review by AlexanderSehr — CHANGES_REQUESTED** [review]
Requires uncommenting in the `/.github/CODEOWNERS` file

File: .github/workflows/avm.res.compute.disk.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    paths:
+      - ".github/actions/templates/avm-**"
+      - ".github/workflows/avm.template.module.yml"
+      - ".github/workflows/avm.res.compute.disk.yml"
+      - "avm/res/compute/disk/**"
+      - "avm/utilities/pipelines/**"
+      - "!*/**/README.md"
+
+env:
+  modulePath: "avm/res/network/network-security-group"
```
Comment (AlexanderSehr) [inline .github/workflows/avm.res.compute.disk.yml:36]:
```suggestion
  modulePath: "avm/res/compute/disk"
```

**Comment by segraef** [comment]
> Requires uncommenting in the `/.github/CODEOWNERS` file

done

File: avm/res/compute/disk/main.bicep [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+@description('Optional. The lock settings of the service.')
+param lock lockType
+
+@description('Optional. Array of role assignments to create.')
+param roleAssignments roleAssignmentType
+
+@description('Optional. Tags of the availability set resource.')
+param tags object?
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
```
Comment (AlexanderSehr) [inline avm/res/compute/disk/main.bicep:130]:
```suggestion
@description('Optional. Enable/Disable usage telemetry for module.')
```

File: avm/res/compute/disk/main.bicep [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+      uploadSizeBytes: createOption == 'Upload' ? uploadSizeBytes : null
+    }
+    diskIOPSReadWrite: contains(sku, 'Ultra') ? diskIOPSReadWrite : null
+    diskMBpsReadWrite: contains(sku, 'Ultra') ? diskMBpsReadWrite : null
+    diskSizeGB: createOption == 'Empty' ? diskSizeGB : null
+    hyperVGeneration: empty(osType) ? null : hyperVGeneration
+    maxShares: maxShares
+    networkAccessPolicy: networkAccessPolicy
+    optimizedForFrequentAttach: optimizedForFrequentAttach
+    osType: empty(osType) ? any(null) : osType
```
Comment (AlexanderSehr) [inline avm/res/compute/disk/main.bicep:193]:
```suggestion
    osType: !empty(osType) ? osType : any(null)
```

File: avm/res/compute/disk/main.bicep [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+      securityDataUri: createOption == 'ImportSecure' ? securityDataUri : null
+      sourceResourceId: createOption == 'Copy' ? sourceResourceId : null
+      sourceUri: createOption == 'Import' ? sourceUri : null
+      storageAccountId: createOption == 'Import' ? storageAccountId : null
+      uploadSizeBytes: createOption == 'Upload' ? uploadSizeBytes : null
+    }
+    diskIOPSReadWrite: contains(sku, 'Ultra') ? diskIOPSReadWrite : null
+    diskMBpsReadWrite: contains(sku, 'Ultra') ? diskMBpsReadWrite : null
+    diskSizeGB: createOption == 'Empty' ? diskSizeGB : null
+    hyperVGeneration: empty(osType) ? null : hyperVGeneration
```
Comment (AlexanderSehr) [inline avm/res/compute/disk/main.bicep:189]:
```suggestion
    hyperVGeneration: !empty(osType) ? hyperVGeneration : null
```

File: avm/res/compute/disk/main.bicep [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+    name: sku
+  }
+  properties: {
+    burstingEnabled: burstingEnabled
+    completionPercent: completionPercent
+    creationData: {
+      createOption: createOption
+      imageReference: createOption != 'FromImage' ? null : {
+        id: imageReferenceId
+      }
```
Comment (AlexanderSehr) [inline avm/res/compute/disk/main.bicep:178]:
```suggestion
      imageReference: createOption == 'FromImage' ? {
        id: imageReferenceId
      } : null
```

File: avm/res/compute/disk/main.bicep [hunk 6]
```diff
@@ -0,0 +1,270 @@
...
+    hyperVGeneration: empty(osType) ? null : hyperVGeneration
+    maxShares: maxShares
+    networkAccessPolicy: networkAccessPolicy
+    optimizedForFrequentAttach: optimizedForFrequentAttach
+    osType: empty(osType) ? any(null) : osType
+    publicNetworkAccess: publicNetworkAccess
+    supportedCapabilities: empty(osType) ? {} : {
+      acceleratedNetwork: acceleratedNetwork
+      architecture: empty(architecture) ? null : architecture
+    }
```
Comment (AlexanderSehr) [inline avm/res/compute/disk/main.bicep:198]:
```suggestion
    supportedCapabilities: !empty(osType) ? {
      acceleratedNetwork: acceleratedNetwork
      architecture: !empty(architecture) ? architecture : null
    } : {}
```

File: avm/res/compute/disk/tests/e2e/import/main.test.bicep [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+        roleDefinitionIdOrName: 'b24988ac-6180-42a0-ab88-20f7382dd24c'
+        principalId: nestedDependencies.outputs.managedIdentityPrincipalId
+        principalType: 'ServicePrincipal'
+      }
+      {
+        roleDefinitionIdOrName: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'acdd72a7-3385-48ef-bd42-f606fba81ae7')
+        principalId: nestedDependencies.outputs.managedIdentityPrincipalId
+        principalType: 'ServicePrincipal'
+      }
+    ]
```
Comment (AlexanderSehr) [inline avm/res/compute/disk/tests/e2e/import/main.test.bicep:75]:
```suggestion
```

File: avm/res/compute/disk/tests/e2e/import/main.test.bicep [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+    sourceUri: nestedDependencies.outputs.vhdUri
+    storageAccountId: nestedDependencies.outputs.storageAccountResourceId
+    tags: {
+      'hidden-title': 'This is visible in the resource name'
+      Environment: 'Non-Prod'
+      Role: 'DeploymentValidation'
+    }
```
Comment (AlexanderSehr) [inline avm/res/compute/disk/tests/e2e/import/main.test.bicep:82]:
```suggestion
```

File: avm/res/compute/disk/tests/e2e/image/main.test.bicep [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+        roleDefinitionIdOrName: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'acdd72a7-3385-48ef-bd42-f606fba81ae7')
+        principalId: nestedDependencies.outputs.managedIdentityPrincipalId
+        principalType: 'ServicePrincipal'
+      }
+    ]
+    tags: {
+      'hidden-title': 'This is visible in the resource name'
+      Environment: 'Non-Prod'
+      Role: 'DeploymentValidation'
+    }
```
Comment (AlexanderSehr) [inline avm/res/compute/disk/tests/e2e/image/main.test.bicep:77]:
```suggestion
```

File: .github/workflows/avm.res.compute.disk.yml [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+
+concurrency:
+  group: ${{ github.workflow }}
+
+jobs:
+  ###########################
+  #   Initialize pipeline   #
+  ###########################
+  job_initialize_pipeline:
+    runs-on: ubuntu-20.04
```
Comment (eriqua) [inline .github/workflows/avm.res.compute.disk.yml:47]:
```suggestion
    runs-on: ubuntu-latest
```

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**